tests/main/boot-state/task.yaml

```
@@ -0,0 +1,77 @@
+summary: smoke test for the boot-state tool
+
+execute: |
+
+    # Check help
+    "$TESTSTOOLS"/boot-state | MATCH "usage: bootenv"
+    "$TESTSTOOLS"/boot-state -h | MATCH "usage: bootenv"
+    "$TESTSTOOLS"/boot-state --help | MATCH "usage: bootenv"
+
+    # check the bootenv command
+    case "$SPREAD_SYSTEM" in
+        ubuntu-core-16-*|ubuntu-core-18-*)
+            # check snap_core and snap_kernel vars are set in bootnev
+            "$TESTSTOOLS"/boot-state bootenv show | MATCH 'snap_core=core.*.snap'
+            "$TESTSTOOLS"/boot-state bootenv show | MATCH 'snap_kernel=.*-kernel_.*.snap'
+            ;;
+        ubuntu-core-20-*)
+            # check kernel_status var is set in bootnev
+            "$TESTSTOOLS"/boot-state bootenv show | MATCH 'kernel_status='
+            ;;
+        *)
+            # check bootnev command can be called
+            "$TESTSTOOLS"/boot-state bootenv show
+            ;; 
+    esac
+
+    # check the get-boot-path command
+    case "$SPREAD_SYSTEM" in
+        ubuntu-core-16-arm-*|ubuntu-core-18-arm-*|ubuntu-core-20-arm-*)
+            "$TESTSTOOLS"/boot-state boot-path | MATCH "/boot/uboot/"
+            ;;
+        fedora-*|opensuse-*|amazon-*|centos-*)
+            "$TESTSTOOLS"/boot-state boot-path | MATCH "/boot/grub2/"
+            ;;
+        *)
+            "$TESTSTOOLS"/boot-state boot-path | MATCH "/boot/grub/"
+            ;;
+    esac
+
+    # check a new var can be set in bootenv
+    "$TESTSTOOLS"/boot-state bootenv show | NOMATCH 'snap_test_1='
+    "$TESTSTOOLS"/boot-state bootenv set snap_test_1 test_1
+    "$TESTSTOOLS"/boot-state bootenv show | MATCH 'snap_test_1=test_1$'
+
+    # check a var can be showed in bootenv
+    "$TESTSTOOLS"/boot-state bootenv set snap_test_2 test_2
+    "$TESTSTOOLS"/boot-state bootenv show snap_test_1 | MATCH 'test_1'
+    "$TESTSTOOLS"/boot-state bootenv show snap_test_1 | NOMATCH 'test_2'
+
+    # check a var can be set even if it is already defined in bootenv
+    "$TESTSTOOLS"/boot-state bootenv set snap_test_1 test_3
+    "$TESTSTOOLS"/boot-state bootenv show snap_test_1 | MATCH 'test_3'
+    test "$("$TESTSTOOLS"/boot-state bootenv show | grep -c snap_test_1)" -eq 1
+
+    # check an existing var can be unset in bootenv
+    "$TESTSTOOLS"/boot-state bootenv unset snap_test_1
+    "$TESTSTOOLS"/boot-state bootenv show | NOMATCH 'snap_test_1='
+    "$TESTSTOOLS"/boot-state bootenv show | MATCH 'snap_test_2=test_2$'
+    "$TESTSTOOLS"/boot-state bootenv unset snap_test_2
+    "$TESTSTOOLS"/boot-state bootenv show | NOMATCH 'snap_test_2='
+
+    # check an unexisting var can be unset in bootenv
```

I would phrase this differently:
```suggestion
    # check an inexistent var can be unset in bootenv
```

---

done